File: app/src/main/java/com/example/miui_ify/ViewPagerAdapter.java
package com.example.miui_ify;

import android.app.Fragment;
import android.app.FragmentManager;
import android.content.Context;

import androidx.annotation.NonNull;
import androidx.legacy.app.FragmentPagerAdapter;

import com.example.settingapp.util.SharePref;

public class ViewPagerAdapter extends FragmentPagerAdapter {
    public static Context mContext;
    int num;
    int du;
    public static int numpager ;
    private Context context;
    public ViewPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return new IconSettingFragment();
    }

    @Override
    public int getCount() {
        num = SharePref.getIntPref(mContext, "total_tile") / (SharePref.getIntPref(mContext, "number_column") * SharePref.getIntPref(mContext, "number_row"));
        du = SharePref.getIntPref(mContext, "total_tile") % (SharePref.getIntPref(mContext, "number_column") * SharePref.getIntPref(mContext, "number_row"));
        if (du != 0) {
            numpager = num + 1;
            return numpager;
        } else {
            numpager = num;
            return num;
        }

    }
}

File: app/src/main/java/com/example/settingapp/welcome/ViewPagerAdapter.kt
package com.example.settingapp.welcome

import android.widget.ImageView
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.example.settingapp.R

class ViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
    var oneFragment : OneFragment?=null
    var twoFragment : TwoFragment?=null
    var threeFragment : ThreeFragment?=null
    var FourFragment : FourFragment?=null
    var FiveFragment : FiveFragment?=null
    var img_view : ImageView?=null
    override fun getItem(position: Int): Fragment {
        when (position) {
            0 -> {
                oneFragment= OneFragment();
                return oneFragment as OneFragment
                img_view!!.setImageResource(R.drawable.ic_arrow_forward)
            }
            1 -> {
                twoFragment= TwoFragment();
                return twoFragment as TwoFragment

            }
            2 -> {
                threeFragment= ThreeFragment();
                return threeFragment as ThreeFragment

            }
            3 -> {
                FourFragment= FourFragment();
                return FourFragment as FourFragment

            }
            4 -> {
                FiveFragment= FiveFragment();
                return FiveFragment as FiveFragment
            }
            else ->{
                oneFragment= OneFragment()
                return oneFragment as OneFragment

            }
        }
        throw IllegalStateException("No fragment at position $position")
    }

    override fun getCount(): Int {
        return 5
    }
}